grails-app/jobs/org/pih/warehouse/jobs/RefreshProductAvailabilityJob.groovy

```diff
@@ -35,30 +31,7 @@ class RefreshProductAvailabilityJob {
             Object productIds = context.mergedJobDataMap.get("productIds")
             String locationId = context.mergedJobDataMap.get("locationId")
 
-            // Calculate product availability for a single location/product, or all products within a single location
-            if (locationId) {
-                Location location = Location.load(locationId)
-                if (productIds && locationId) {
-                    productIds.each { productId ->
-                        Product product = Product.load(productId)
-                        productAvailabilityService.refreshProductAvailability(location, product, forceRefresh)
-                    }
-                }
-                else {
-                    productAvailabilityService.refreshProductAvailability(location, forceRefresh)
-                }
-            }
-            // Calculate product availability for a single product within all locations
-            else if (productIds) {
-                productIds.each { productId ->
-                    Product product = Product.load(productId)
-                    productAvailabilityService.refreshProductAvailability(product, forceRefresh)
-                }
-            }
-            // Calculate product availability for all products within all locations
-            else {
-                productAvailabilityService.refreshProductAvailability(forceRefresh)
-            }
+            productAvailabilityService.refreshProductsAvailability(locationId, productIds, forceRefresh)
```

Comment:
We need to be careful with this though. The method signature implies that we're updating all products for a given location. But there's a side effect in that if we pass in a null locationId then we'll be updating these products over all locations. Which might not be what we want in all cases.